grafana/grafana
# Usage insights query caching

**What this PR does / why we need it**:

This adds `totalQueries` and `cachedQueries` counts to the usage insights event logs. Admins want to know how many queries are being satisfied by the cache, since caching has many benefits. See the issue for more info.

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana-enterprise/issues/2639

**How to test**

- enable caching for one of your datasources
- enable usage insights and usage insights export storage in your `custom.ini`
- make a few queries to your datasource
- explore your usage logs that were exported. All `data-request` logs should have `totalQueries` and `cachedQueries` fields present.
- All `dashboard-request` logs will not have those fields present
